From 42ec3fc4e367fcb6e67887dfa1b65e7e2db1a133 Mon Sep 17 00:00:00 2001
From: Zhang Xiao <xiao.zhang@windriver.com>
Date: Fri, 23 Oct 2020 01:22:15 -0700
Subject: [PATCH] net: phy: aquantia: fix a bug of function
 aquantia_config_advert

In checking "advertising" settings with linkmode_test_bit, the
bit number should be used instead of another bitmap.

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 drivers/net/phy/aquantia_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/phy/aquantia_main.c b/drivers/net/phy/aquantia_main.c
index 7c99cefc7b4b..e4aeb510c550 100644
--- a/drivers/net/phy/aquantia_main.c
+++ b/drivers/net/phy/aquantia_main.c
@@ -406,9 +406,9 @@ static int aquantia_config_advert(struct phy_device *phydev)
 
 	if (linkmode_test_bit(ETHTOOL_LINK_MODE_1000baseT_Full_BIT, phydev->advertising))
 		adv1 |= 0x8000;
-	if (linkmode_test_bit(ADVERTISED_10000baseT_Full, phydev->advertising))
+	if (linkmode_test_bit(ETHTOOL_LINK_MODE_10000baseT_Full_BIT, phydev->advertising))
 		adv |= 0x1000;
-	if (linkmode_test_bit(ADVERTISED_2500baseX_Full, phydev->advertising))
+	if (linkmode_test_bit(ETHTOOL_LINK_MODE_2500baseX_Full_BIT, phydev->advertising))
 		adv1 |= 0x400;
 
 	if (adv != oldadv) {
-- 
2.17.1

